File: lib/models/drug_result.dart
class DrugResult {
  final String? rxcui;
  final String name;
  final String? dosage;
  final String? form;
  final String source;
  final double confidence;

  const DrugResult({
    this.rxcui,
    required this.name,
    this.dosage,
    this.form,
    required this.source,
    required this.confidence,
  });

  factory DrugResult.fromJson(Map<String, dynamic> json) {
    return DrugResult(
      rxcui: json['rxcui'] as String?,
      name: json['name'] as String,
      dosage: json['dosage'] as String?,
      form: json['form'] as String?,
      source: json['source'] as String,
      confidence: (json['confidence'] as num).toDouble(),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'rxcui': rxcui,
      'name': name,
      'dosage': dosage,
      'form': form,
      'source': source,
      'confidence': confidence,
    };
  }
}

File: PillChecker/Models/DrugResult.swift
import Foundation

struct DrugResult: Codable, Identifiable, Hashable {
    let rxcui: String?
    let name: String
    let dosage: String?
    let form: String?
    let source: String
    let confidence: Double
    let needsConfirmation: Bool?

    var id: String { rxcui ?? name }

    enum CodingKeys: String, CodingKey {
        case rxcui, name, dosage, form, source, confidence
        case needsConfirmation = "needs_confirmation"
    }
}
